libs/js-sdk/src/client.ts

```diff
@@ -110,45 +115,66 @@ export function createBrowserClient({
             })
         },
         subscribeToInitialisedChanged(callback) {
-            initialisedCallbacks.push(callback)
+            debugLog('Subscribing to initialised changed: %o', {
+                initialised: isInitialised(),
+            })
+
+            initialisedState.initialisedCallbacks.push(callback)
             return () => {
-                initialisedCallbacks.splice(
-                    initialisedCallbacks.indexOf(callback),
+                initialisedState.initialisedCallbacks.splice(
+                    initialisedState.initialisedCallbacks.indexOf(callback),
                     1,
                 )
             }
         },
         async updateAudiences(updatedAudiences: string[]) {
+            if (compareArrays(state.audiences, updatedAudiences)) {
+                debugLog('Skipped updating audiences, no change: %o', {
+                    updatedAudiences,
+                    currentAudiences: state.audiences,
+                    initialised: isInitialised(),
+                })
+                // No need to update audiences
+                return Promise.resolve()
+            }
+
             debugLog('Updating audiences: %o', {
                 updatedAudiences,
+                currentAudiences: state.audiences,
                 initialised: isInitialised(),
             })
 
-            if (compareArrays(state.audiences, updatedAudiences)) {
-                // No need to update audiences
-                return Promise.resolve()
-            }
+            const newPromise = new PromiseCompletionSource<boolean>()
+            initialisedState.initialisedPromise = newPromise
+            initialisedState.initialisedPromise.promise.catch(() => {})
+            initialisedState.initialisedPromise.promise.then(() => {
```

Comment:
Rather than calling the initialised callbacks on a specific resolution code path, listen to promise completion and call then as long as the promise is still the same instance.